### ISSUE TITLE
Bomb and Star in Advanced Example

### ISSUE THREAD

**Issue by dcrawl** [britzl/emthree#7]
When a Bomb blows up a Starfish, the game seems to stop working.

**Comment by JCash** [comment]
"the game seems to stop working."

Can you describe it a bit more?

**Comment by dcrawl** [comment]
As soon as the Bomb was matched, it detonated clearing several fish around it until it hit the starfish.  Then the game stopped.  The game pieces didn't fall and the game froze.  I'm not sure but I think even the bubbles from the fish stopped moving.  It was like it didn't know what todo when the Starfish was removed.  Maybe it didn't know what colour to choose for the starfish to activate?  Other than that, it just locked up.

**Comment by deckarep** [comment]
@dcrawl - I’ve been using this library for a few weeks now and it’s been working pretty much flawlessly for me. 

It would help if you can provide more details about the environment you are on and what OS. Also did you see any errors in the Defold logging?

Is this issue with the Ocean or Basic example or your own game implementation?

Also, can you replicate the lockup? If so, that would be helpful for the maintainers to help you. I can also try to help.

**Comment by dcrawl** [comment]
I was playing the Ocean example at:  https://britzl.github.io/Emthree/index.html .  So I don't have access to any logs.  I'm using an M1 MacBook Air with Safari as the browser.  Everything is fully patched.  I'm not running it in Defold.  

To replicate the lockup, I would have to play until I had a Starfish (straight line match of 5) and a bomb (T shaped match of 5) within range of each other and set off the bomb (with a match of 3).  It was an uncommon situation that happened after playing for a while.  Probably longer than 20 minutes.

**Comment by dcrawl** [comment]
I managed to replicate it.  It does seem to stop working as soon as the Bomb explodes.  The animations do keep running but I can't interact with the fish anymore.  They wont move.  There is also an empty space on the board where the bomb blew up  a couple of fish and then the starfish.  As soon as the starfish is removed, it stops working.  See the image attached.

![image](https://github.com/britzl/emthree/assets/4784891/028bc28d-dda6-4346-81c0-82ab55c4482c)

**Comment by britzl** [comment]
There's actually some keybindings which lets you set up the board in a certain way:

<img width="459" alt="Screenshot 2023-07-07 at 08 16 08" src="https://github.com/britzl/emthree/assets/1300688/2ab79a37-5627-4fbc-a7b4-ae7018028998">

**Comment by britzl** [comment]
Maybe you could provide a screenshot of a board setup that causes this behaviour?

**Comment by dcrawl** [comment]
This is a Mockup of what the board looked like:
![image](https://github.com/britzl/emthree/assets/4784891/d7ee9299-7947-4068-a83a-0af086f464b0)
If I moved the Bomb to the left to make a match with the two other blowfish, then the explosion catches the Starfish and the game freezes.

**Comment by britzl** [comment]
I'm still not able to reproduce the issue. Does it happen all the time?

https://github.com/britzl/emthree/assets/1300688/6d4d0db3-9745-49cf-87dd-56be315bf59c

**Comment by dcrawl** [comment]
It's happened twice in a row for me.  Is the Itch web demo up to date?  Perhaps it's old?

**Comment by britzl** [comment]
> It's happened twice in a row for me. Is the Itch web demo up to date? Perhaps it's old?

Perhaps, but I don't think so. I've updated it just in case and I will try to reproduce it again.

**Comment by britzl** [comment]
I guess it might matter which color the bomb is (yes it has a color, which is used if it is destroyed as a side effect of some other special block).

**Comment by dcrawl** [comment]
I'll try to reproduce it again, myself but I'll see if I can film it this time.

EDIT:  I'm getting a 404 now at that URL.  I guess you took it down.

**Comment by britzl** [comment]
Ah, sorry, messed up the upload. Please try again!

**Comment by dcrawl** [comment]
I've got it on candid camera this time.  In fact, I found a second potential issue.  At around 40 seconds in, there is a match made on a Horizontal Bomb that would produce another Horizontal Bomb.  But the first Bomb does not trigger.  It just creates a new Bomb.

At the end of the video is where another Bomb, blows up the Star and freezes the game.

https://github.com/britzl/emthree/assets/4784891/fcf7944e-6709-4a57-9c55-e8bb6d00a2db
